apps/evm/src/clients/api/mutations/withdraw/useWithdraw.ts

```diff
@@ -3,19 +3,20 @@ import withdraw, { type WithdrawInput } from 'clients/api/mutations/withdraw';
 import FunctionKey from 'constants/functionKey';
 import { type UseSendTransactionOptions, useSendTransaction } from 'hooks/useSendTransaction';
 import { useAnalytics } from 'libs/analytics';
-import { useGetVTokenContract } from 'libs/contracts';
+import { useGetNativeTokenGatewayContract, useGetVTokenContract } from 'libs/contracts';
 import { useChainId } from 'libs/wallet';
-import type { VToken } from 'types';
+import type { Pool, VToken } from 'types';
 import { callOrThrow, convertMantissaToTokens } from 'utilities';
 
 type TrimmedRedeemInput = Omit<WithdrawInput, 'tokenContract' | 'accountAddress'>;
 type Options = UseSendTransactionOptions<TrimmedRedeemInput>;
 
-const useWithdraw = (
-  { vToken, poolName }: { vToken: VToken; poolName: string },
```

Comment:
Same as for `useWithdraw`, I'd keep the `poolName` prop instead of requiring to pass the whole `pool` object.